### ISSUE TITLE
[Feature Request]: Allow `KryptonNavigator` to support form minimise, maximise/restore and close `ButtonSpec` buttons

### ISSUE THREAD

**Issue by PWagner1** [Krypton-Suite/Standard-Toolkit#927]
Whilst investigating a solution to #925, I have noticed that the `KryptonNavigator` does not currently support  form minimise, maximise/restore and close ButtonSpec buttons.

![image](https://user-images.githubusercontent.com/949607/215263009-7085f066-682e-4247-910a-5c37f613acbc.png)

**Comment by Smurf-IV** [comment]
So what are these ?
![image](https://user-images.githubusercontent.com/2418812/215263197-c904e6dc-8140-46dc-aac8-245e23570416.png)

**Comment by Wagnerp** [comment]
> So what are these ? ![image](https://user-images.githubusercontent.com/2418812/215263197-c904e6dc-8140-46dc-aac8-245e23570416.png)

They only control the `KryptonPage`(s) inside of the control

**Comment by Smurf-IV** [comment]
> They only control the `KryptonPage`(s) inside of the control


Which is where the control will be displayed up against the top of the border?

![image](https://user-images.githubusercontent.com/2418812/215263346-b31df314-50a6-48d8-9e01-2990898ba67a.png)

**Comment by Wagnerp** [comment]
Only in the tabs section, unsure if you can just put the close button on the tab though...

**Comment by Smurf-IV** [comment]
Yupe:
![image](https://user-images.githubusercontent.com/2418812/215263538-3942916a-d73a-4029-9a85-5b9b054a570b.png)

**Comment by Wagnerp** [comment]
> Yupe: ![image](https://user-images.githubusercontent.com/2418812/215263538-3942916a-d73a-4029-9a85-5b9b054a570b.png)

Might be an idea to set this option as a default?

**Comment by Wagnerp** [comment]
I was looking in the wrong place!!!

![image](https://user-images.githubusercontent.com/949607/215263682-38f65f70-59d3-4395-b918-1d11d0dd178e.png)

**Comment by Wagnerp** [comment]
@Smurf-IV Updated the title topic to better describe the issue

**Comment by sdreb3421** [comment]
> 

I didn't know you could do that, how do you do per tab buttons?

**Comment by Smurf-IV** [comment]
> > 
> 
> I didn't know you could do that, how do you do per tab buttons?

Best to check out the "Navigator" Demo's
![image](https://user-images.githubusercontent.com/2418812/215985808-d3039edb-46ac-48e4-9959-6f0caae8bbcc.png)

**Comment by Wagnerp** [comment]
@Smurf-IV The current iteration isn't going to work, because it needs access to `internal` & `private` fields that are in `Krypton.Toolkit`. So, the only option is to either assign a custom KCommand to these buttonspecs to do such operations _or_ introduce a 'new' KForm property to navigator to hook up the Close etc options. The latter might be the easiest. Thoughts?

**Comment by Smurf-IV** [comment]
Either make them public,
or use the "friends internals" equivalent attribute to allow krypton dll's to access other krypton dll's internal classes (That just prevents "Some" optimisation, it's no biggy otherwise !).

**Comment by Wagnerp** [comment]
I'll look into it this week

**Comment by Wagnerp** [comment]
Am I missing something here?

![image](https://user-images.githubusercontent.com/949607/218306010-546c1af4-552c-4588-96a8-a876979ccf8b.png)

**Comment by Smurf-IV** [comment]
> Am I missing something here?

Yes.. That means it is at Global scope, rather than just at a single class.
I normally put it into (and directly above) the class definition that is being "Allowed" to be visible from _whom_

**Comment by Wagnerp** [comment]
Found a workaround, but I need to find the public key of the DLL

**Comment by Smurf-IV** [comment]
Using 'sn.exe' utility:
> `sn -T YourAssembly.dll`

**Comment by Wagnerp** [comment]
> Using 'sn.exe' utility:

Yep, this could also resolve #943 as well

**Comment by Wagnerp** [comment]
Yes!!!!!!

![image](https://user-images.githubusercontent.com/949607/218307631-52fe183f-f4ed-4cb2-a4de-2fbc9659f275.png)

**Comment by Smurf-IV** [comment]
@PWagner1 What is the state of this please, It seems to go round in circles a bit ??

**Comment by PWagner1** [comment]
> @PWagner1 What is the state of this please, It seems to go round in circles a bit ??

@Smurf-IV I think it's still outstanding
